### ISSUE TITLE
Filter for file extension more than 3 letters seems not working on Mac OS X

### ISSUE THREAD

**Issue by nepluno** [samhocevar/portable-file-dialogs#72]
Hi Sam, thanks for your effort on this amazing repo.

I'm trying to use open file dialog to select Json files (*.json) on Mac OS X (Monterey 12.3) and the dialog just show them all in grey. The generated OSX script 
`osascript -e "set ret to choose file default location \".\" with prompt \"Load Json\" of type {\"json\"}"`
seems to be fine but even executing it in a terminal the files are still grey and cannot be selected. 

This issue also applies to other files with extension that's more than 3 letters (e.g., I've tried filtering *.aaaa or *.bbbb and none of them works, but filtering something like .jso or .js just works).

It seems this could be an OSX issue but is there a workaround?

**Comment by samhocevar** [comment]
Hi! And thanks for spotting this problem. I believe I found a workaround: if the list starts with a three-letter extension, everything works again. Prepending `"///"` appears to work fine, so I pushed that fix to 6b36ebf4 if you want to test it.

Note that the bug is really weird:
 - `{"txt"}` works as expected
 - `{"json"}` does not work (everything is greyed out)
 - `{"txt","json"}` works as expected
 - `{"json","txt"}` does not work (now everything can be selected!)
